lib/util.ts

```diff
@@ -489,17 +489,23 @@ export class ShapeUtil {
         } else {
           reshapedDims[i] = shapeHints[i];
         }
-        size *= reshapedDims[i];
+        newTensorSize *= reshapedDims[i];
       }
     }
 
+    const oldTensorSize = ShapeUtil.size(originalDims);
     if (unknownDimension !== -1) {
-      const originalTensorFlattenedSize = ShapeUtil.size(originalDims);
-      if (originalTensorFlattenedSize % size !== 0) {
+      if (oldTensorSize % newTensorSize !== 0) {
         throw new Error(`the input tensor cannot be reshaped to the requested shape. Input shape: [${
             originalDims}] Output shape: [${shapeHints}]`);
       }
-      reshapedDims[unknownDimension] = originalTensorFlattenedSize / size;
+      reshapedDims[unknownDimension] = oldTensorSize / newTensorSize;
+    }
+    // validate sizes from originalDims and reshapedDims match
+    else {
+      if (newTensorSize !== oldTensorSize) {
```

Comment:
Adding check here to ensure the sizes of the originalDims and reshapedDims remain the same after reshape.